File: MasterCatalog.Domain/Models/ItemAttributeOption.cs
using System.ComponentModel.DataAnnotations;

namespace MasterCatalog.Domain.Models
{
    public class ItemAttributeOption
    {
        public int? ItemAttributeOptionID { get; set; }

        public int ItemAttributeID { get; set; }

        [Required(AllowEmptyStrings = false, ErrorMessage = "Option value is required")]
        public string AttributeOption { set; get; }
    }
}

File: MasterCatalog.Web/wwwroot/js/models/itemAttributeOption.js
export class ItemAttributeOption {
    itemAttributeOptionID;
    itemAttributeID;
    attributeOption;
    attributeOptionError;

    constructor(id, attributeId, option) {
        this.itemAttributeOptionID = id;
        this.itemAttributeID = attributeId;
        this.attributeOption = option;
    }

    validate() {
        this.clearErrors();

        let valid = true;
        const option = this.attributeOption;
        if (!option) {
            this.attributeOptionError = 'Option is Required.';
            valid = false;
        }

        return valid;
    }

    clearErrors() {
        this.attributeOptionError = null;
    }
}